libraries/AP_RangeFinder/AP_RangeFinder_MaxsonarSerialIRXL.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2016  Intel Corporation. All rights reserved.
+ *
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AP_RangeFinder_MaxsonarSerialIRXL.h"
+
+#include <AP_HAL/AP_HAL.h>
+#include <ctype.h>
+
+#define MAXSONAR_SERIAL_LV_BAUD_RATE 9600
+
+extern const AP_HAL::HAL& hal;
+
+// RXXXX\r XXXX: 0 padding 4 digits cm. 
+// read - return last value measured by sensor
```

Comment:
this isn't true ... you are doing the average

Reply:
I'll change.
I will write the following specifications
Spec:
The output is an ASCII capital “R”, followed by four ASCII character digits representing the range in centimeters, followed by a carriage return (ASCII 13). The maximum distance reported is 1650.